flytekit/loggers.py

```diff
@@ -17,77 +18,129 @@
-user_space_logger = child_loggers["user_space"]
-
-# create console handler
-if os.environ.get(LOGGING_RICH_FMT_ENV_VAR) != "0":
-    try:
-        handler = RichHandler(
-            rich_tracebacks=True,
-            omit_repeated_times=False,
-            keywords=["[flytekit]"],
-            log_time_format="%Y-%m-%d %H:%M:%S,%f",
-            console=Console(width=os.get_terminal_size().columns),
-        )
-    except OSError:
-        handler = logging.StreamHandler()
-else:
+
+def set_flytekit_log_properties(
+    handler: typing.Optional[logging.Handler] = None,
+    filter: typing.Optional[logging.Filter] = None,
+    level: typing.Optional[int] = None,
+):
+    """
+    flytekit logger, refers to the framework logger. It is possible to selectively tune the logging for flytekit.
+
+    Sets the flytekit logger to the specified handler, filter, and level. If any of the parameters are None, then
+    the corresponding property on the flytekit logger will not be set.
+
+    :param handler: logging.Handler to add to the flytekit logger
+    :param filter: logging.Filter to add to the flytekit logger
+    :param level: logging level to set the flytekit logger to
+    """
+    global logger
+    if handler is not None:
+        logger.handlers.clear()
+        logger.addHandler(handler)
+    if filter is not None:
+        logger.addFilter(filter)
+    if level is not None:
+        logger.setLevel(level)
+
+
+def set_user_logger_properties(
+    handler: typing.Optional[logging.Handler] = None,
+    filter: typing.Optional[logging.Filter] = None,
+    level: typing.Optional[int] = None,
+):
+    """
+    user_space logger, refers to the user's logger. It is possible to selectively tune the logging for the user.
+
+    :param handler: logging.Handler to add to the user_space_logger
+    :param filter: logging.Filter to add to the user_space_logger
+    :param level: logging level to set the user_space_logger to
+    """
+    global user_space_logger
+    if handler is not None:
+        user_space_logger.addHandler(handler)
+    if filter is not None:
+        user_space_logger.addFilter(filter)
+    if level is not None:
+        user_space_logger.setLevel(level)
+
+
+def _get_env_logging_level() -> int:
+    """
+    Returns the logging level set in the environment variable, or logging.WARNING if the environment variable is not
+    set.
+    """
+    return int(os.getenv(LOGGING_ENV_VAR, logging.WARNING))
+
+
+def initialize_global_loggers():
+    """
+    Initializes the global loggers to the default configuration.
+    """
     handler = logging.StreamHandler()
+    handler.setLevel(logging.DEBUG)
+    formatter = logging.Formatter(fmt="[%(name)s] %(message)s")
+    if os.environ.get(LOGGING_FMT_ENV_VAR, "json") == "json":
+        formatter = jsonlogger.JsonFormatter(fmt="%(asctime)s %(name)s %(levelname)s %(message)s")
+    handler.setFormatter(formatter)
 
-handler.setLevel(logging.DEBUG)
```

Comment:
This is unfortunately a breaking change because we use this `handler` for further customization. This is the second time this part breaks :(

Reply:
@honnix , sorry for the breaking change. The intent of this change was to simplify the logging infrastructure. 

Can you talk about what sort of customizations you are applying to this? We might simply expose that as an escape hatch.

Reply:
No worries. We have adapted accordingly. So there are two needs:

* we need to customize the json formatter to remap a field (levelname) to satisfy stackdriver logging 
* we use the same handler to set all existing loggers (things from other libs) so we can make sure everything is json logged, not just flytekit and user space loggers